### PR TITLE
Map interaction refacto

### PR DESCRIPTION
**Pull request by nerik**
This PR aims at leaving interaction priority choices outside `map-components`: refactors interaction actions to not give priority to legacy heatmap layers, but instead gives a maximum of data about all features below map cursor to callbacks.

The signature of the event object sent to module callbacks has been cleaned up and redesigned.

Additionally, an `autoClusterZoom` option has been added to the map module.

This is a breaking changes and will require adaptations in `map-client` and optionally in in `data-portal`

### CODE REVIEW

File: src/map/glmap/viewport.actions.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ const transitionTo = (increment, latitude = null, longitude = null, zoom = null)
 }
 
 export const transitionToZoom = (zoom) => (dispatch) => {
+  debugger
```
Comment (j8seangel) [inline src/map/glmap/viewport.actions.js]: 🙄

File: src/map/config.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const MAX_SPRITES_PER_LAYER = 200000
 // ie how large the clicked region should be for including vessels
 export const VESSEL_CLICK_TOLERANCE_PX = 10
 
-export const CLUSTER_CLICK_ZOOM_INCREMENT = 1
+export const CLUSTER_CLICK_ZOOM_INCREMENT = 2
```
Comment (j8seangel) [inline src/map/config.js]: Does mapbox provide a zoom level where the cluster is going to "expand"? It could be useful instead of this fixed value

Comment (j8seangel) [reply src/map/config.js]: I saw you could get it with https://docs.mapbox.com/mapbox-gl-js/api/#geojsonsource#getclusterexpansionzoom and it is executed in the code but doesn't seem to be used later.

File: src/map/README.md [hunk 3]
```diff
@@ -297,11 +305,39 @@ Function. TODO
 
 ## `onClick`
 
-Function. TODO
+Function. Returns an event:
```
Comment (j8seangel) [inline src/map/README.md]: Nice we have the documentation updated!

File: src/map/README.md [hunk 4]
```diff
@@ -297,11 +305,39 @@ Function. TODO
...
+                        id:string,
+                        label:string,
+                        value:any,
+                        title:string,
+                        isLink:bool
+                    }
+                ]
+            }
+        ],
+        feature: {} - set if there's only one feature
```
Comment (j8seangel) [inline src/map/README.md]: We have duplicated feature, is this something legacy to maintain and keep compatibility? if not I'd remove it to keep only the array of features

File: src/map/glmap/viewport.actions.js (hunk 1 above)
Comment (nerik) [reply src/map/glmap/viewport.actions.js]: 😞

File: src/map/config.js (hunk 2 above)
Comment (nerik) [reply src/map/config.js]: Good catch, thanks. Addressed here: https://github.com/GlobalFishingWatch/map-components/pull/57/commits/30d3b018692f9ef22483bb7d5ee110a444dcca28

File: src/map/README.md (hunk 4 above)
Comment (nerik) [reply src/map/README.md]: You're right: https://github.com/GlobalFishingWatch/map-components/pull/57/commits/30d3b018692f9ef22483bb7d5ee110a444dcca28#diff-69b34ee7f05dfc863ca8b45572c8827bR335

**Comment by j8seangel** [comment]
great job!

**Pull request merged**